syncserver/src/db/mysql/models.rs

```diff
@@ -489,7 +489,7 @@ impl MysqlDb {
                 bso::expiry,
             ))
             .filter(bso::user_id.eq(user_id))
-            .filter(bso::collection_id.eq(collection_id as i32)) // XXX:
+            .filter(bso::collection_id.eq(collection_id)) // XXX:
```

Comment:
Can you kill these while you're here, they were about the casting.

```suggestion
            .filter(bso::collection_id.eq(collection_id))
```